packages/asset-swapper/src/utils/rfq_maker_blacklist.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Tracks a maker's history of timely responses, and manages whether a given
+ * maker should be avoided for being too latent.
+ */
+export class RfqMakerBlacklist {
+    // tslint:disable-next-line:custom-no-magic-numbers
+    private static readonly _msPerMinute = 1000 * 60;
+    private readonly _makerTimeoutStreakLength: { [makerUrl: string]: number } = {};
+    private readonly _makerBlacklistedUntilDate: { [makerUrl: string]: number } = {};
+    constructor(private readonly _blacklistDurationMinutes: number, private readonly _timeoutStreakThreshold: number) {}
+    public logTimeoutOrLackThereof(makerUrl: string, didTimeout: boolean): void {
+        if (!this._makerTimeoutStreakLength.hasOwnProperty(makerUrl)) {
+            this._makerTimeoutStreakLength[makerUrl] = 0;
+        }
+        if (didTimeout) {
+            this._makerTimeoutStreakLength[makerUrl] += 1;
+            if (this._makerTimeoutStreakLength[makerUrl] > this._timeoutStreakThreshold) {
+                this._makerBlacklistedUntilDate[makerUrl] =
+                    Date.now() + this._blacklistDurationMinutes * RfqMakerBlacklist._msPerMinute;
+                this._makerTimeoutStreakLength[makerUrl] = 0;
+            }
+        } else {
+            this._makerTimeoutStreakLength[makerUrl] = 0;
+        }
+    }
+    public isMakerBlacklisted(makerUrl: string): boolean {
+        if (this._makerBlacklistedUntilDate.hasOwnProperty(makerUrl)) {
+            if (this._makerBlacklistedUntilDate[makerUrl] > Date.now()) {
```

Comment:
what is the difference between `new Date().getTime()` and `Date.now()`? we usually use the former afaik

Reply:
No difference in effect, really.  I prefer `Date.now()`, partly because in theory it's less overhead (no constructing a `Date` object), but more importantly because it's easy to tell at a glance what the intention is, without needing to remember that `new Date()` constructs an object set to `"now"`.